docs/examples/examples.rst

```diff
@@ -9,4 +9,5 @@ Examples
    Reading and writing a Lance dataset in Rust <./write_read_dataset.rst>
    Creating Multi-Modal datasets using Lance <./flickr8k_dataset_creation.rst>
    Training Multi-Modal models using a Lance dataset <./clip_training.rst>
-   Deep Learning Artefact Management using Lance <./artefact_management.rst>
\ No newline at end of file
+   Deep Learning Artefact Management using Lance <./artefact_management.rst>
+   Reading and writing a Lance dataset in Spark <./spark_example.rst>
```

Comment:
I think this is a spark data source example. It will have another spark catalog connector example when the catalog is ready.
So this doc should rename as `spark_datasource_example.rst`